ports/nrf/common-hal/neopixel_write/__init__.c

```diff
@@ -116,6 +116,8 @@ void common_hal_neopixel_write (const digitalio_digitalinout_obj_t* digitalinout
 
     uint32_t pattern_size = PATTERN_SIZE(numBytes);
     uint16_t* pixels_pattern = NULL;
+    static uint16_t* pixels_pattern_heap = NULL;
```

Comment:
Please make sure that this is reset to NULL when the heap is reset otherwise it'll break after a reload.

Reply:
@tannewt where can I find the heap reset code?

Reply:
I'd add a `neopixel_write_reset` and add it here: https://github.com/adafruit/circuitpython/blob/master/main.c#L190

Reply:
I added the neopixel_write_reset to the port only, as this is only for nrf.  I called it from reset_port().

Reply:
Feel free to stub it into other ports if that is simpler. I wouldn't be surprised if others needed it in the future.